src/riscv/lib/src/machine_state/page_cache/interpreted.rs

```diff
@@ -25,6 +26,17 @@ use crate::state_backend::ManagerWrite;
 #[derive(Debug, Default, Clone)]
 pub struct InterpretedCompiler;
 
+/// This implementation of [`RouterEq`] keeps the ranges in the [`Router`] as small as possible, to
+/// reduce the number of pages that get unneccessarily dropped.
```

Comment:
```suggestion
/// Since `InterpretedCompiler` does no compilation at all, [`router_eq`] will always return false, so that the ranges in the [`Router`] are kept as small as possible, which helps
/// reduce the number of pages that could get unnecessarily dropped.
```

nit, maybe something like this? (ie there's no need to say 'this implementation', since it's a comment on an implementation